bin/kano-settings-onboot

```diff
@@ -360,172 +158,132 @@ def ensure_correct_volume_levels():
 
-    # If the hdmi_group and hdmi_mode values are set explicitly then skip the detection.
-    if get_config_value('hdmi_group', ignore_comments=True) != 0 and \
-       get_config_value('hdmi_mode', ignore_comments=True) != 0:
-
-        logger.info('is comment kano_screen_used? {}'.format(has_config_comment('kano_screen_used')))
-        logger.info('is comment kano_screen_used({})? {}'.format(model, get_config_comment('kano_screen_used', model)))
+    # Apply any corrections to the EDID data and set if screen is monitor.
+    override_models(edid, model)
+    calculate_is_monitor(edid)
 
-        if not force_mode and (not has_config_comment('kano_screen_used') or
-                                   get_config_comment('kano_screen_used', model)):
+    # Fix HDMI audio status.
+    if not edid['hdmi_audio'] and is_HDMI():
+        logger.info('HDMI audio not supported on this screen, changing to analogue.')
+        if not dry_run:
+            set_to_HDMI(False)
 
-            # The screen is either the same or not set at all
-            logger.info('Explicit HDMI configuration detected, exiting.')
-            sys.exit(1)
-        else:
-            logger.info('New screen was detected, reconfiguring...')
+    # Calculate changes needed...
+    changes = compare_and_set_full_range(edid, status, model, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    # override modes, calculate is_monitor
-    override_models(edid, model)
-    calculate_is_monitor(edid)
+    changes = compare_and_set_overscan(edid, status, model, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    if verbose:
-        # Explain the new screen settings
-        print 'New status:', json.dumps(status, indent=4)
-        print 'New EDID:', json.dumps(edid, indent=4)
+    changes = compare_and_set_optimal_resolution(edid, status, supported_modes, dry_run=dry_run, force=force_mode)
+    reboot_required = reboot_required or changes
 
-    # fix hdmi audio status
-    if not edid['hdmi_audio'] and is_HDMI():
-        logger.info('hdmi audio not supported on this screen, changing to analogue')
-        set_to_HDMI(False)
+    return (reboot_required, RC_SUCCESS)
 
-    # calculate changes needed
-    overall_changes = False
 
-    changes = compare_and_set_mode(edid, status)
-    overall_changes = overall_changes or changes
+def calculate_is_monitor(edid):
+    """
+    TODO: What is this for?
+    """
+    edid['target_full_range'] = edid['is_monitor']
 
-    changes = compare_and_set_full_range(edid, status)
-    overall_changes = overall_changes or changes
 
-    changes = compare_and_set_overscan(edid, status)
-    overall_changes = overall_changes or changes
+def reboot(dry_run=False):
+    """
+    Flush logs to disk and reboot the kit.
+    """
+    if not dry_run:
+        logger.sync()
+        run_cmd('kano-checked-reboot changes systemctl reboot --force')
```

Comment:
Are we sure about this one? this `sync` simply flushes the logger buffers, not the filesystem.
Specially delicate with a `--force` reboot...

 * https://github.com/KanoComputing/kano-toolset/blob/master/kano/_logging.py#L214

Reply:
I see your point here @skarbat. The only reasons I'd be reluctant to change it is 3 things:

1. This preserves the functionality we had before and didn't have problems.
2. The `end_config_transaction` right above the call here already does an `os.system('sync')`.
3. It would slow it down even more - Why do we even need to tell the OS to flush the filesystem in the first place?

Also, as to why we're using `--force` there is beyond me...

Reply:
Oh okay, I wasn't aware of point 2). Your question about `--force` was because requesting a reboot in the middle of a bootup would be asking for troubles.